x/epochstorage/keeper/migrations.go

```diff
@@ -116,64 +119,146 @@ func (m Migrator) Migrate3to4(ctx sdk.Context) error {
+	// 	}
 
-		store.Set(iterator.Key(), m.keeper.cdc.MustMarshal(&stakeStorageV6))
-	}
+	// 	store.Set(iterator.Key(), m.keeper.cdc.MustMarshal(&stakeStorageV6))
+	// }
 
 	return nil
 }
 
 // Migrate6to7 goes over all existing stake entries and populates the new description field with current moniker
 func (m Migrator) Migrate6to7(ctx sdk.Context) error {
-	utils.LavaFormatDebug("migrate: epochstorage to include detailed description")
+	// utils.LavaFormatDebug("migrate: epochstorage to include detailed description")
 
-	store := prefix.NewStore(ctx.KVStore(m.keeper.storeKey), types.KeyPrefix(types.StakeStorageKeyPrefix))
+	// store := prefix.NewStore(ctx.KVStore(m.keeper.storeKey), types.KeyPrefix(types.StakeEntriesPrefix))
+	// iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	// defer iterator.Close()
+
+	// for ; iterator.Valid(); iterator.Next() {
+	// 	var stakeStorageV7 types.StakeStorage
+	// 	m.keeper.cdc.MustUnmarshal(iterator.Value(), &stakeStorageV7)
+
+	// 	for i := range stakeStorageV7.StakeEntries {
+	// 		stakeStorageV7.StakeEntries[i].Description.Moniker = stakeStorageV7.StakeEntries[i].Moniker
+	// 		stakeStorageV7.StakeEntries[i].Moniker = ""
+	// 	}
+
+	// 	store.Set(iterator.Key(), m.keeper.cdc.MustMarshal(&stakeStorageV7))
+	// }
+
+	return nil
+}
+
+// Migrate7to8 transfers all the stake entries from the old stake storage to the new stake entries store
+// StakeStorage is set to the stake entries store
+// StakeStorageCurrent is set to the stake entries current store
+// StakeStorageUnstake is deleted
+func (m Migrator) Migrate7to8(ctx sdk.Context) error {
+	utils.LavaFormatDebug("migrate: epochstorage to move stake entries from stake storage")
+	k := m.keeper
+
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), v3.KeyPrefix(v3.StakeStorageKeyPrefix))
 	iterator := sdk.KVStorePrefixIterator(store, []byte{})
 
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
-		var stakeStorageV7 types.StakeStorage
-		m.keeper.cdc.MustUnmarshal(iterator.Value(), &stakeStorageV7)
+		key := string(iterator.Key())
 
-		for i := range stakeStorageV7.StakeEntries {
-			stakeStorageV7.StakeEntries[i].Description.Moniker = stakeStorageV7.StakeEntries[i].Moniker
-			stakeStorageV7.StakeEntries[i].Moniker = ""
+		// identify stake storage type: regular, current or unstake
+		if key == v3.StakeStorageKeyUnstakeConst {
+			store.Delete(iterator.Key())
+			continue
 		}
 
-		store.Set(iterator.Key(), m.keeper.cdc.MustMarshal(&stakeStorageV7))
+		epoch, err := extractEpochFromStakeStorageKey(key)
+		isCurrentStakeStorage := m.isCurrentStakeStorageKey(ctx, key)
+		if err != nil && !isCurrentStakeStorage {
+			panic(fmt.Errorf("stake storage with unidentified index %s", key))
+		}
+
+		var stakeStorage types.StakeStorage
+		k.cdc.MustUnmarshal(iterator.Value(), &stakeStorage)
+
+		for _, entry := range stakeStorage.StakeEntries {
+			if isCurrentStakeStorage {
```

Comment:
what about the unstake stake storage?
we should make sure we dont save them

Reply:
the unstake stake storage has a single const key. On line 41, I check if the stake storage key is of the unstake stake storage and delete it.